src/site/content/en/blog/web-animations/index.md

```diff
@@ -0,0 +1,262 @@
+
+Within CSS, this would be cumbersome to recreate, especially when applying unique, yet sequenced animations to multiple elements. You'd have to use a `"@keyframe"`, sort out the correct timing percentages to place the animations, and use `"animation-delay"` prior to triggering the animations in the sequence.
+
+## Play, pause, and reverse
+
+What can open, should close! Luckily, since [Chrome 39](https://developers.google.com/web/updates/2014/12/web-animation-playback), the WAAPI has provided us the ability to play, pause, and reverse our animations.
+
+You can take the above animation, and give it a smooth, reversed animation when clicking the button again using `".reverse()"`. This way, you can create a smoother and more contextual interaction for our modal.
+
+<figure class="w-figure">
+  <video controls autoplay loop muted class="w-screenshot">
+    <source src="https://storage.googleapis.com/web-dev-assets/web-animations/modal-reverse.mp4">
+  </video>
+  <figcaption class="w-figcaption">
+    An example of a modal opening and closing upon button click. <a href="https://codepen.io/kevers-google/details/LYpvdNG">See Demo on Codepen </a>
+  </figcaption>
+</figure>
+
+What you can do is create two play-pending animations (`"openModal"`, and an inline opacity transformation), and then pause one of the animations, delaying it until the other is finished. You can then use promises to wait for each to be finished before playing. Finally, you can check to see if a flag is set, and then reverse each animation.
+
+```js/17-20
+document.querySelector('button').addEventListener('click', () => {                                 
+  // Create two play-pending animations.
+  let animations = [
+    document.querySelector('.modal').animate(openModal, openModalSettings),
+    document.querySelector('.text').animate({ opacity: [0, 1] }, openModalSettings)
+  ];
+
+  // Pause one of the animations and delay it until the other is finished.
+  if (open) {
+    animations.forEach((anim) => {
+      anim.currentTime = anim.effect.getTiming().duration;
+      anim.reverse();
+    });
+    animations = animations.reverse();
+  }
+  
+  // Run animations sequentially.
+  animations[1].pause();  
+  animations[0].finished.then(() => {
+    animations[1].play();
+  });
+  open = !open;
+});
+```
+
+## Dynamic interactions with partial keyframes
+
+<figure class="w-figure">
+<video controls autoplay loop muted class="w-screenshot">
+    <source src="https://storage.googleapis.com/web-dev-assets/web-animations/retargetting.mp4">
+  </video>  <figcaption class="w-figcaption">
+    Retargeting example, where a mouse click adjusts the animation to a new location. <a href="https://codepen.io/una/details/BaoveQJ">See Demo on Codepen </a>
+  </figcaption>
+</figure>
```

Comment:
Please embed as a Glitch rather than linking to Codepen: https://web.dev/handbook/web-dev-components/#glitches